### DISCUSSION TITLE
Support of Iroh with surrealdb and substrate blockchain

### DISCUSSION THREAD

**Discussion by silicology1** [n0-computer/iroh#1454] [Ideas]
Surrealdb has separation of storage and api layers. It would be interesting to have Iroh as the [storage solution](https://surrealdb.com/docs/embedding) for Surrealdb. Also possibility of emending iroh and Surrealdb in [substrate blockchain](https://github.com/integritee-network/worker/issues/1435#issue-1878925010) for offchain computing.

**Comment by b5** [comment]
Super cool @amiyatulu. I'd never heard of surreal db, really cool to see such a polished rust product, I'll need more time to understand if / how an iroh integration would work. If I understand correctly your high level is **do computation on encrypted data in a substrate sidechain** Is that right?

Seems like we'd have to do a bunch of things to get that to work. off the top of my head you'd need:

* [ ] support iroh / surrealDB blob loading / unloading to do offchain data handling in the integritee-network worker processes
* [ ] implement an encryption solution to store private data on iroh
* [ ] have a means of granting permission to a worker process in integritee nodes, such that they can decrypt data
* [ ] (probably) provide a means for writing an encrypted data result back to iroh

That sounds like a tall order, but if you're interested in pursuing this, we can help 😄, starting by making sure I have the requirements right

**Comment by silicology1** [comment]
@b5 Yes, that's what I need. Currently, I don't need encrypted data for my current project, but it's required for future projects where private data is used. I am interested in pursuing this.
